File: src/main/java/com/flexpoker/chat/service/ChatService.java
package com.flexpoker.chat.service;

import java.util.UUID;

public interface ChatService {

    void saveAndPushSystemLobbyChatMessage(String message);

    void saveAndPushUserLobbyChatMessage(String message, String username);

    void saveAndPushSystemGameChatMessage(UUID gameId, String message);

    void saveAndPushUserGameChatMessage(UUID gameId, String message, String username);

    void saveAndPushSystemTableChatMessage(UUID gameId, UUID tableId, String message);

    void saveAndPushUserTableChatMessage(UUID gameId, UUID tableId, String message, String username);

}

File: src/main/kotlin/com/flexpoker/chat/service/ChatService.kt
package com.flexpoker.chat.service

import java.util.UUID

interface ChatService {
    fun saveAndPushSystemLobbyChatMessage(message: String)
    fun saveAndPushUserLobbyChatMessage(message: String, username: String?)
    fun saveAndPushSystemGameChatMessage(gameId: UUID?, message: String)
    fun saveAndPushUserGameChatMessage(gameId: UUID?, message: String, username: String?)
    fun saveAndPushSystemTableChatMessage(gameId: UUID?, tableId: UUID?, message: String)
    fun saveAndPushUserTableChatMessage(gameId: UUID?, tableId: UUID?, message: String, username: String?)
}